pkg/commands/oscommands/os.go

```diff
@@ -108,13 +108,29 @@ func (c *OSCommand) Quote(message string) string {
 // AppendLineToFile adds a new line in file
 func (c *OSCommand) AppendLineToFile(filename, line string) error {
 	c.LogCommand(fmt.Sprintf("Appending '%s' to file '%s'", line, filename), false)
-	f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0o600)
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_RDWR|os.O_CREATE, 0o600)
 	if err != nil {
 		return utils.WrapError(err)
 	}
 	defer f.Close()
 
-	_, err = f.WriteString("\n" + line)
+	info, err := os.Stat(filename)
+	if err != nil {
+		return utils.WrapError(err)
+	}
+
+	// if the last byte is newline, append newline as well
```

Comment:
I actually don't think we should care whether the user has a newline there already: I can't imagine anybody complaining if we add one ourselves regardless of what pattern was there before. It's generally considered good practice to end files in a newline to ensure a clean diff when somebody else goes and adds a new line to it

Reply:
> EDIT: I'm dumb: this actually does make sense

Well now I'm not sure what to think anymore. It makes no difference to me whether there's a newline at the end or not, and I haven't heard anyone getting bitten by this, but if you have, I'll gladly just chuck `"\n + line + \n"` at the end of the file and call it a day.